pkg/reconciler/adapter.go

```diff
@@ -309,3 +317,22 @@ func MaybeAppendValueFromEnvVar(envs []corev1.EnvVar, key string, valueFrom v1al
 
 	return envs
 }
+
+// adapterOverrideOptions applies adapter override parameters depending on
+// deployment type.
+func adapterOverrideOptions(overrides *v1alpha1.AdapterOverrides) resource.ObjectOption {
+	knServiceVisibility := resource.VisibilityClusterLocal
+	if overrides.Public {
+		knServiceVisibility = resource.VisibilityPublic
+	}
```

Comment:
Echoing a previous comment: we probably want `overrides.Public` to be optional, otherwise `kubectl get` will always show it, even if users never set it.

And I would also move those bits to the `case *servingv1.Service` because that's the only case where this parameter is relevant.

That logic could become something like:

```golang
func adapterOverrideOptions(overrides *v1alpha1.AdapterOverrides) []resource.ObjectOption {
	var opts []resource.ObjectOption // assumes we are now returning individual options

	if _, ok := object.(*servingv1.Service); ok {
		if pub := overrides.Public; pub != nil && pub {
			append(opts, resource.VisibilityPublic)
		}
	}

	if overrides.Resources != nil {
		// do the Resources magic ...
	}
}
```